pkg/urbit/jets/b/flop.c

```diff
@@ -3,46 +3,29 @@
 */
 #include "all.h"
 
+u3_noun
+u3qb_flop(u3_noun a)
+{
+  u3_noun i, t = a, b = u3_nul;
 
-/* functions
-*/
-  u3_noun
-  u3qb_flop(u3_noun a)
-  {
-    u3_noun b = 0;
-
-    while ( 1 ) {
-      if ( u3_nul == a ) {
-        return b;
-      }
-      else if ( c3n == u3du(a) ) {
-        u3z(b);
-
-        return u3m_bail(c3__exit);
-      }
-      else {
-        b = u3nc(u3k(u3h(a)), b);
-        a = u3t(a);
-      }
-    }
+  while ( u3_nul != t ) {
+    u3x_cell(t, &i, &t);
```

Comment:
Calling u3x_cell with an out pointer of its argument is well-defined right? Bc it's a pointer passed by-value

Reply:
Right, the noun reference (first argument) is passed by value.